libs/domain/checkout/composition/src/composition.component.ts

```diff
@@ -1,127 +1,41 @@
+import { html, LitElement, TemplateResult } from 'lit';
+import { CheckoutCompositionComponentOptions } from './composition.model';
 import { compositionStyles } from './composition.styles';
 
+@defaultOptions({
+  checkoutAuthPage: '/checkout-login',
+})
 @hydratable('window:load')
-export class CheckoutCompositionComponent extends ComponentMixin<CheckoutCompositionOptions>() {
-  static styles = compositionStyles;
-
-  protected orchestrationService = resolve(CheckoutOrchestrationService);
-
-  protected options$ = new ContentController(this).getOptions();
-
-  protected isEmptyCart$ = resolve(CartService).isEmpty();
-
-  protected isAuthenticated$ = resolve(AuthService).isAuthenticated();
-  protected hasAddresses$ = resolve(AddressService)
-    .getAddresses()
-    .pipe(map((addresses) => !!addresses?.length));
-  protected isGuestCheckout$ = resolve(CheckoutDataService).isGuestCheckout();
-
-  protected steps$ = this.orchestrationService.getValidity();
-
-  protected checkout$ = combineLatest([
-    this.isEmptyCart$,
-    this.isAuthenticated$,
-    this.isGuestCheckout$,
-    this.hasAddresses$,
-    this.steps$,
-    this.options$,
-  ]);
-
-  protected override render(): TemplateResult {
-    return html` ${asyncValue(
-      this.checkout$,
-      ([
-        isEmptyCart,
-        isAuthenticated,
-        isGuestCheckout,
-        hasAddresses,
-        steps,
-        options,
-      ]) => {
-        if (isEmptyCart) {
-          return html``;
-        }
-
-        return html`
-          <checkout-auth .options=${{ disableGuest: options.disableGuest }}>
-          </checkout-auth>
-          ${when(
-            isAuthenticated || isGuestCheckout,
-            () =>
-              html`
-                ${steps.map(({ id }, index) => {
-                  return html`${this.renderStep(
-                    id,
-                    this.renderHeading(index, id, isAuthenticated, hasAddresses)
-                  )}`;
-                })}
-              `
-          )}
-        `;
-      }
-    )}`;
-  }
-
-  protected renderStep(
-    step: CheckoutStepType,
-    heading: TemplateResult
-  ): TemplateResult {
-    let content: TemplateResult = html``;
-
-    switch (step) {
-      case CheckoutStepType.Delivery:
-        content = html` <checkout-delivery></checkout-delivery>`;
-        break;
-      case CheckoutStepType.Shipping:
-        content = html`<checkout-shipment></checkout-shipment>`;
-        break;
-      case CheckoutStepType.Payment:
-        content = html`<checkout-payment></checkout-payment>`;
-        break;
+export class CheckoutCompositionComponent extends CheckoutMixin(
+  ContentMixin<CheckoutCompositionComponentOptions>(LitElement)
+) {
+  static styles = [compositionStyles];
+
+  protected requiresRedirect = computed(
+    () =>
+      this.$options().checkoutAuthPage &&
+      !this.isGuest() &&
+      !this.isAuthenticated()
+  );
+
+  connectedCallback(): void {
+    if (this.requiresRedirect()) {
```

Comment:
I had it first on an effect, but it was too aggressive, as it would kick in at:
- after placing the order
- when you use the ordinary logout
in both cases we'd land at checkout-login